src/clipping/sutherlandhodgman.jl

```diff
@@ -14,6 +14,49 @@ The Sutherland-Hodgman algorithm for clipping polygons.
 """
 struct SutherlandHodgman <: ClippingMethod end
 
-function clip(poly::Polygon, other::Geometry, ::SutherlandHodgman)
-  @error "not implemented"
+function clip(poly::Polygon, other::Geometry, method::ClippingMethod) 
+  c = [clip(ring, boundary(other), method) for ring in rings(poly)]
+  r = filter(!isnothing, c)
+
+  if isempty(r)
+    nothing
+  elseif hasholes(poly)
+    PolyArea(r[1], r[2:end])
+  else
+    PolyArea(r[1])
+  end
+end
```

Comment:
Given that his function is method-agnostic, please move it to the src/clipping.jl file where other methods will be included later.